samples/matter/lock/src/credentials/credentials_data_types.h

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <app/clusters/door-lock-server/door-lock-server.h>
+#include <lib/core/ClusterEnums.h>
+
+namespace DoorLockData
+{
+
+static constexpr size_t kMaxCredentialLength = CONFIG_LOCK_MAX_CREDENTIAL_LENGTH;
+
+/**
+ * @brief Packing to buffer
+ *
+ * Given `data` with size `dataSize` will be packed to the `buff` and the `offset` will be
+ * increased by `dataSize` bytes.
+ *
+ * This function expects that `buff` and `data` pointers are valid and not NULL.
+ *
+ */
+void pack(void *buff, const void *data, size_t dataSize, size_t &offset);
+
+/**
+ * @brief Unpacking from buffer to object
+ *
+ * `dataSize` amount of bytes from `buff` will be unpacked to the `data` and the `offset` will be
+ * increased by `dataSize` bytes.
+ *
+ * This function expects that `buff` and `data` pointers are valid and not NULL.
+ *
+ */
+void unpack(void *data, size_t dataSize, const void *buff, size_t &offset);
+
+struct Credential {
+	union Info {
+		struct __attribute__((__packed__)) Fields {
+			uint8_t mStatus;
+			uint8_t mCredentialType;
+			uint8_t mCreationSource;
+			uint8_t mCreatedBy;
+			uint8_t mModificationSource;
+			uint8_t mLastModifiedBy;
+		} mFields;
+		uint8_t mRaw[sizeof(mFields)];
+	};
+	static_assert(sizeof(Info) == 6);
+
+	struct Secret {
+		size_t mDataLength;
+		uint8_t mData[kMaxCredentialLength];
+	};
+
+	Info mInfo;
+	Secret mSecrets;
```

Comment:
Super minor - as it's not a list or array it should be singular?
```suggestion
	Secret mSecret;
```